tests/components/vacuum/test_demo.py

```diff
@@ -0,0 +1,189 @@
+        """Test if methods call the services as expected."""
+        self.hass.states.set(ENTITY_VACUUM_BASIC, STATE_ON)
+        self.assertTrue(vacuum.is_on(self.hass, ENTITY_VACUUM_BASIC))
+
+        self.hass.states.set(ENTITY_VACUUM_BASIC, STATE_OFF)
+        self.assertFalse(vacuum.is_on(self.hass, ENTITY_VACUUM_BASIC))
+
+        self.hass.states.set(ENTITY_ID_ALL_VACUUMS, STATE_ON)
+        self.assertTrue(vacuum.is_on(self.hass))
+
+        self.hass.states.set(ENTITY_ID_ALL_VACUUMS, STATE_OFF)
+        self.assertFalse(vacuum.is_on(self.hass))
+
+        vacuum.turn_on(self.hass, ENTITY_VACUUM_COMPLETE)
+        self.hass.block_till_done()
+        self.assertTrue(vacuum.is_on(self.hass, ENTITY_VACUUM_COMPLETE))
+
+        vacuum.turn_off(self.hass, ENTITY_VACUUM_COMPLETE)
+        self.hass.block_till_done()
+        self.assertFalse(vacuum.is_on(self.hass, ENTITY_VACUUM_COMPLETE))
+
+        vacuum.toggle(self.hass, ENTITY_VACUUM_COMPLETE)
+        self.hass.block_till_done()
+        self.assertTrue(vacuum.is_on(self.hass, ENTITY_VACUUM_COMPLETE))
+
+        vacuum.start_pause(self.hass, ENTITY_VACUUM_COMPLETE)
+        self.hass.block_till_done()
+        self.assertFalse(vacuum.is_on(self.hass, ENTITY_VACUUM_COMPLETE))
+
+        vacuum.start_pause(self.hass, ENTITY_VACUUM_COMPLETE)
+        self.hass.block_till_done()
+        self.assertTrue(vacuum.is_on(self.hass, ENTITY_VACUUM_COMPLETE))
+
+        vacuum.stop(self.hass, ENTITY_VACUUM_COMPLETE)
+        self.hass.block_till_done()
+        self.assertFalse(vacuum.is_on(self.hass, ENTITY_VACUUM_COMPLETE))
+
+        state = self.hass.states.get(ENTITY_VACUUM_COMPLETE)
+        self.assertLess(state.attributes.get(ATTR_BATTERY_LEVEL), 100)
+        self.assertNotEqual("Charging", state.attributes.get(ATTR_STATUS))
+
+        vacuum.locate(self.hass, ENTITY_VACUUM_COMPLETE)
+        self.hass.block_till_done()
+        state = self.hass.states.get(ENTITY_VACUUM_COMPLETE)
+        self.assertIn("I'm over here", state.attributes.get(ATTR_STATUS))
+
+        vacuum.return_to_base(self.hass, ENTITY_VACUUM_COMPLETE)
+        self.hass.block_till_done()
+        state = self.hass.states.get(ENTITY_VACUUM_COMPLETE)
+        self.assertIn("Returning home", state.attributes.get(ATTR_STATUS))
+
+        vacuum.set_fan_speed(self.hass, FAN_SPEEDS[-1],
+                             entity_id=ENTITY_VACUUM_COMPLETE)
+        self.hass.block_till_done()
+        state = self.hass.states.get(ENTITY_VACUUM_COMPLETE)
+        self.assertEqual(FAN_SPEEDS[-1], state.attributes.get(ATTR_FAN_SPEED))
+
+    def test_services(self):
+        """Test vacuum services."""
+        # Test send_command
+        send_command_calls = mock_service(
+            self.hass, DOMAIN, SERVICE_SEND_COMMAND)
+
+        params = {"rotate": 150, "speed": 20}
+        vacuum.send_command(
+            self.hass, entity_id=ENTITY_VACUUM_BASIC,
+            command='test_command', params=params)
```

Comment:
`command` is not a keyword argument for the `send_command` function. I'm surprised that this works!?